File: Haskell/Atividade-5/exercicio-10.hs
{-
Crie uma funcao com assinatura filtrar :: (t -> Bool) -> [t] -> [t], a qual recebe uma funcao, uma lista e retorna uma nova lista. 
Esta funcao aplica a funcao recebida como parametro sobre cada elemento da lista e caso o retorno da funcao for verdadeiro, 
entao o elemento fara parte da nova lista, caso contrario nao. Para esta tarefa, utilize o conceito de list comprehension. 
-}

filtrar :: (t -> Bool) -> [t] -> [t]
filtrar f lista = [x | x <- lista, f x]

ehPar :: Int -> Bool
ehPar x = mod x 2 == 0

main = do
  print (filtrar ehPar [1..10])

File: Prolog/Atividade-16/exercicio-10.pl
/*
Crie uma regra inverte(L1,L2), a qual recebe uma lista L1 como parametro e deve retornar a mesma invertida L2.
*/

concatena([],L,L).
concatena([H|T],L2,[H|L3]) :- concatena(T,L2,L3).

inverte([],[]).
inverte([H|T],L2) :- inverte(T,LT), concatena(LT,[H],L2).
